### PR TITLE
chore: Refactor feature view modules

### PR DESCRIPTION
**Pull request by sudohainguyen**
**What this PR does / why we need it**:
Related to #3361 
This PR is the very first step to conduct refactoring the entire `feast` repository by focusing on feature view modules.
I have moved all the implementation to more well-structured sub-modules under `core` module.

**Changes Summary**:
- `base_feature_view.py` -> `core/feature_views/base.py`
- `on_demand_feature_view.py` -> `core/feature_views/on_demand_fv.py`
- `feature_view.py` -> `core/feature_views/generic_fv.py`
- `stream_feature_view.py` -> `core/feature_views/stream_fv.py`
- `batch_feature_view.py` -> `core/feature_views/batch_fv.py`
- `request_feature_view.py` -> `core/feature_views/request_fv.py`

### CODE REVIEW

**Comment by achals** [comment]
Hi @sudohainguyen! Thanks for your efforts in this area - I agree that there's room for improvement in our code organization. However, I'm afraid of making such an invasive change since it's changes the public API that all users rely on and will require code changes by all users. 

At a high level, is there a specific problem, or something that we can't currently do, that requires this reorganization?

**Comment by sudohainguyen** [comment]
> Hi @sudohainguyen! Thanks for your efforts in this area - I agree that there's room for improvement in our code organization. However, I'm afraid of making such an invasive change since it's changes the public API that all users rely on and will require code changes by all users.
> 
> At a high level, is there a specific problem, or something that we can't currently do, that requires this reorganization?

hi @achals thanks for your comment.
these are just my suggestions and I already made some backward compatibility adjustments. So that it wouldn't affect the current workflow.

as a minor contributor, I see quite hard to follow code structure, I believe others might feel the same. If core maintainers don't think it's neccessary, I would love to close the work 😄

**Comment by sudohainguyen** [comment]
all tests passed
hi @achals what do you think? should we proceed?

**Comment by ketangangal** [comment]
It would be great if you guys can merge this pull request.
